### PR TITLE
[fix] - dagster-deltalake: make storage backend options return a string dict

### PR DESCRIPTION
**Pull request by marijncv**
## Summary & Motivation
The current way that `DeltaTableResource` passes configuration values to the `DeltaTable` constructor breaks when passing a dict that have valuetypes other than str (e.g. int, bool or None) with the error:
```python
E       TypeError: argument 'storage_options': 'NoneType' object cannot be converted to 'PyString'
```
This happens even when the user does not pass any of those value types, because there are default values of these types.

To replicate, try:
```python
import pyarrow as pa
from deltalake import write_deltalake
from dagster_deltalake.config import AzureConfig
data = pa.table(
    {
        "a": pa.array([1, 2, 3], type=pa.int32()),
        "b": pa.array([5, 6, 7], type=pa.int32()),
    }
)
write_deltalake("table", data, storage_options=AzureConfig(account_name="test").dict())
```

This PR adds a `str_dict()` method to the config classes to make sure they return string values.

## How I Tested These Changes
Added unit tests in `python_modules/libraries/dagster-deltalake/dagster_deltalake_tests/test_delta_table_resource.py`

### CODE REVIEW

File: python_modules/libraries/dagster-deltalake/dagster_deltalake_tests/test_delta_table_resource.py [hunk 1]
```diff
@@ -62,3 +63,24 @@ def read_table(delta_table: DeltaTableResource):
...
+)
+def test_resource_storage_options(tmp_path, config):
+    data = pa.table(
+        {
+            "a": pa.array([1, 2, 3], type=pa.int32()),
+            "b": pa.array([5, 6, 7], type=pa.int32()),
+        }
+    )
+
+    write_deltalake(os.path.join(tmp_path, "table"), data, storage_options=config.dict())
```
Comment (mlarose) [inline python_modules/libraries/dagster-deltalake/dagster_deltalake_tests/test_delta_table_resource.py:86]: hi, I don't understand how is this testing that the returned value of .dict() matches what you intend? is there an expectation that write_deltalake does that validation properly or throws an exception?

Comment (marijncv) [reply python_modules/libraries/dagster-deltalake/dagster_deltalake_tests/test_delta_table_resource.py:86]: Hi @mlarose, thanks for the comment. Indeed the expectation is that by passing the config to `storage_options` in `write_deltalake` it validates that the config is passed in such a way that write_deltalake does not throw an exception.

I will make a small addition to do more explicit validation of the dict. Let me know what you think!

**Review by mlarose — APPROVED** [review]
it looks good to me.

@roeap another PR if you have the time, your integration is popular it seems.

**Review by roeap — COMMENTED** [review]
@marijncv thanks for fixing this! @mlarose LGTM!

**Pull request merged**